### PR TITLE
Fix str of RayTaskError

### PR DESCRIPTION
**Pull request by ericl**
## Why are these changes needed?

Fix the str representation -- some exceptions like KeyError override str, so we have to override that with the right class order.

### CODE REVIEW

**Review by richardliaw — APPROVED** [review]

**Comment by mitchellstern** [comment]
This will need to be merged with the (non-conflicting) changes to `python/ray/exceptions.py` from https://github.com/ray-project/ray/pull/5857/files#diff-1e1d82facc9e441a27ef4b4ae602490a.

File: python/ray/exceptions.py [hunk 1]
```diff
@@ -58,24 +61,22 @@ def as_instanceof_cause(self):
         if issubclass(RayTaskError, self.cause_cls):
             return self  # already satisfied
 
-        class cls(self.cause_cls, RayTaskError):
-            def __init__(self, function_name, traceback_str, cause_cls, pid,
-                         host):
+        class cls(RayTaskError, self.cause_cls):
+            def __init__(self, function_name, traceback_str, cause_cls,
+                         proctitle, pid, host):
```
Comment (mitchellstern) [inline python/ray/exceptions.py]: We should change `host` to `ip`.

File: python/ray/exceptions.py [hunk 2]
```diff
@@ -58,24 +61,22 @@ def as_instanceof_cause(self):
...
                 RayTaskError.__init__(self, function_name, traceback_str,
-                                      cause_cls, pid, host)
+                                      cause_cls, proctitle, pid, host)
```
Comment (mitchellstern) [inline python/ray/exceptions.py]: We should change `host` to `ip`.

**Review by mitchellstern — APPROVED** [review]

**Comment by ericl** [comment]
@simon-mo looks like serve tests are flaky

**Comment by simon-mo** [comment]
@ericl should be fixed by #5886

**Pull request merged**